src/ActionMenu2.tsx

```diff
@@ -90,5 +87,9 @@ const MenuButton = React.forwardRef<AnchorRef, ButtonProps>((props, anchorRef) =
   )
 })
 
+/** this component is syntactical sugar 🍭 */
+type MenuOverlayProps = Partial<OverlayProps> & {children: React.ReactElement[] | React.ReactElement}
+const Overlay: React.FC<MenuOverlayProps> = props => <>{props.children}</>
```

Comment:
Hmm, I think I understand what's going on here, but I wonder if it'll be confusing that...
```
<ActionMenu>
  <ActionMenu.Overlay>
  ...
  </ActionMenu.Overlay>
</ActionMenu>
```

... will give you the same result as... 
```
<ActionMenu>
   ...
</ActionMenu>
```

In Autocomplete, rendering without the Overlay gives you a different result: https://primer.style/react/Autocomplete#example-without-autocompleteoverlay

Reply:
Oh yeah, this is a (probably evil) first draft. I want to move most of that Menu logic to MenuOverlay if it's possible.

Reply:
Why do we have this as an example?: https://primer.style/react/Autocomplete#example-without-autocompleteoverlay 🤔 

Do we see a use case for this?